src/nspawn/nspawn-mount.c

```diff
@@ -703,17 +707,18 @@ static int parse_mount_bind_options(const char *options, unsigned long *mount_fl
         return 0;
 }
 
-static int mount_bind(const char *dest, CustomMount *m) {
+static int mount_bind(const char *dest, CustomMount *m, uid_t uid_shift, uid_t uid_range) {
         _cleanup_free_ char *mount_opts = NULL, *where = NULL;
         unsigned long mount_flags = MS_BIND | MS_REC;
         struct stat source_st, dest_st;
         int r;
+        bool idmapped = false;
```

Comment:
even if its a bit of a compat break  i think we should really default to this behaviour if the kernel supports it, it makes so much more sense.

Reply:
Hmm but how would you do that? There is no easy way to check for support and it depends on the source file system. For example currently (well, actualy ~2 months ago, not sure if it changed but the point stays the same) you can not use filesystem id mapping if the source is a tmpfs.
We could default to true since it does not result in a crash or error if there is no support but just no id mapping. Actually checking the source filesystem against a list of supported filesystems would be a possibility as well but I do not see enought benefit for that to justify the complexity and maintenance overhead of such a list.